src/orion/algo/axoptimizer.py

```diff
@@ -203,9 +204,14 @@ def suggest(self, num):
                 if self.fidelity_index is not None:
                     # Convert 0-dim arrays into python numbers so their type can
                     # be validated by Ax
-                    parameters[self.fidelity_index] = float(
-                        self.space[self.fidelity_index].high
+                    assert isinstance(
```

Comment:
For now all dimensions are always wrapped in a transformedspace/transformeddimension even if the algo does not need any transformation. When I fixed the Space converter for ConfigSpace this hack here won't be necessary and I'll remove. We have the same hack in many algos that well need to be removed.

Reply:
This assert here doesn't seem to have any value, typing-wise. (The while loop narrows the type to TransformedDimension within itself, and there is also an assert for Fidelity below it).
Plus, the code might change at some point, which will make this assertion false.

Removing the assert seems to me like a good idea, since it's unnecessary, and might prevent an error later on.

Reply:
Oh right, the type is verified in the while loop anyway.

Reply:
Fixed in 72301433ce5ff6d53d8b7b13b09b2f13b730520e